lib/faraday/options.rb

```diff
@@ -48,17 +45,32 @@ def clear
     end
 
     # Public
-    def merge(value)
-      dup.update(value)
+    def merge!(other)
+      other.each do |key, other_value|
+        self_value = self.send(key)
+        sub_options = self.class.options_for(key)
+        if sub_options
+          self_value = sub_options.from(self_value)
```

Comment:
Can't we assume self_value will always be a subclass of `Options` and thus doesn't require to be passed through `#from` class method?
If not, could it be an hash? If so, then we should manage also that case with an `elsif` as we do in `update`.